MAINTAINERS.md

```diff
@@ -28,6 +34,39 @@
 
 [This document](https://github.com/opensearch-project/.github/blob/main/MAINTAINERS.md) explains what maintainers do in this repo, and how they should be doing it. If you're interested in contributing, see [CONTRIBUTING](CONTRIBUTING.md).
 
+## Release Owner
+Release owner is a temporary role for the duration of a given OpenSearch / OpenSearch Dashboard release, it is made concrete by assigning a release ticket to them.  They oversee the release ticket and make sure that it completes.  While tactical release actions should always be a part of the release ticket, see the [template](./.github/ISSUE_TEMPLATE/release_template.md), the purpose of the Release owner is to be responsible for following the release process.
```

Comment:
Release Owner
OpenSearch Dashboard*s* 
Break this up in separate sentences instead of `,`.